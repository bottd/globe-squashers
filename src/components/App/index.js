import React, {useState, useEffect} from 'react';
import {connect} from 'react-redux';
import {fetchImages} from '../../thunks/fetchImages';
import './App.css';

export function App({fetchImages, images, page}) {
  const [initialLoad, setInitialLoad] = useState(false);
  useEffect(() => {
    if(!initialLoad) {
      fetchImages(page);
      setInitialLoad(true);
    }
  });
  return (
    <div className="App">
      <h1>Hey</h1>
    </div>
  );
}

const mapStateToProps = state => ({
  page: state.API.page,
  images: state.images
});

const mapDispatchToProps = dispatch => ({
  fetchImages: (page) => dispatch(fetchImages(page))
});

export default connect(
  mapStateToProps,
  mapDispatchToProps,
)(App);
