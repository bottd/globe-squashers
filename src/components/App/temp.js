import React, {useState, useEffect} from 'react';
import {connect} from 'react-redux';
import {Switch, Route} from 'react-router';
import {fetchImages} from '../../thunks/fetchImages';
import Home from '../Home';
import './App.css';

export function App({fetchImages, images, page}) {
  const [initialLoad, setInitialLoad] = useState(false);
  useEffect(() => {
    if (!initialLoad) {
      fetchImages(page);
      setInitialLoad(true);
    }
  });
  return (
    <div className="App">
      <Switch>
        <Route to='/' component={Home}/>
      </Switch>
    </div>
  );
}

const mapStateToProps = state => ({
  page: state.API.page,
  images: state.images,
});

const mapDispatchToProps = dispatch => ({
  fetchImages: page => dispatch(fetchImages(page)),
});

export default connect(
  mapStateToProps,
  mapDispatchToProps,
)(App);
