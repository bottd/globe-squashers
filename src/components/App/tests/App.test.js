import React from 'react';
import ReactDOM from 'react-dom';
import { mount } from 'enzyme';
import App from '../';

describe('App', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = mount(<App/>);
  });

  it('Should render like snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
