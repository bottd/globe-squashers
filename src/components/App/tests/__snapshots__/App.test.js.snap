// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Should render like snapshot 1`] = `
ReactWrapper {
  Symbol(enzyme.__unrendered__): <App
    fetchImages={[MockFunction]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "fetchImages": [MockFunction],
    },
    "ref": null,
    "rendered": Object {
      "instance": <div
        class="App"
      >
        <h1>
          Hey
        </h1>
      </div>,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <h1>
          Hey
        </h1>,
        "className": "App",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": <h1>
            Hey
          </h1>,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Hey",
          },
          "ref": null,
          "rendered": Array [
            "Hey",
          ],
          "type": "h1",
        },
      ],
      "type": "div",
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "fetchImages": [MockFunction],
      },
      "ref": null,
      "rendered": Object {
        "instance": <div
          class="App"
        >
          <h1>
            Hey
          </h1>
        </div>,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <h1>
            Hey
          </h1>,
          "className": "App",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": <h1>
              Hey
            </h1>,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Hey",
            },
            "ref": null,
            "rendered": Array [
              "Hey",
            ],
            "type": "h1",
          },
        ],
        "type": "div",
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
